mobile/apps/photos/lib/services/home_widget_service.dart

```diff
@@ -182,6 +231,105 @@ class HomeWidgetService {
     }
   }
 
+  Future<bool> _captureFileV2(
+    EnteFile file,
+    String key,
+    String title,
+    String? mainKey,
+  ) async {
+    try {
+      late final ImageProvider imageProvider;
+      const double size = THUMBNAIL_SIZE_V2;
+
+      // For local files (images, videos, live photos), use PhotoManager's high-res thumbnails
+      if (!file.isRemoteFile) {
+        // Request 1024px thumbnail from PhotoManager (efficient for all file types)
+        final Uint8List? thumbnail = await getThumbnailFromLocal(
+          file,
+          size: THUMBNAIL_SIZE_V2.toInt(),
+        );
+
+        if (thumbnail == null) {
+          _logger.warning(
+            "Failed to get high-res thumbnail for V2 widget ${file.displayName}",
+          );
+          return false;
+        }
+
+        // PhotoManager returns thumbnails at requested size
+        imageProvider = MemoryImage(thumbnail);
+      }
+      // For remote images, download full file for best quality
+      else if (file.fileType == FileType.image) {
+        final File? imageFile = await getFileFromServer(file);
+
+        if (imageFile == null) {
+          _logger.warning("Failed to get file for V2 widget ${file.displayName}");
+          return false;
+        }
+
+        // Read and validate in isolate to avoid blocking main thread
+        final Uint8List imageBytes = await compute(
```

Comment:
This will spawn a new isolate. Shouldn't we use already existing pool of isolates that is spawned on app init by the `Computer` package? If it is much easier to spawn a new isolate here instead of using one from the pool, let it be.